database/sqlite/sqlite_functions.c

```diff
@@ -1020,3 +1022,53 @@ void add_migrated_file(char *path, uint64_t file_size)
 
     return;
 }
+
+#define SQL_INS_CHART_LABEL "insert or replace into chart_label " \
+    "(chart_id, source_type, label_key, label_value, date_created) " \
+    "values (@chart, @source, @label, @value, strftime('%s'));"
+
+void sql_store_chart_label(uuid_t *chart_uuid, int source_type, char *label, char *value)
+{
+    sqlite3_stmt *res = NULL;
+    int rc;
+
+    rc = sqlite3_prepare_v2(db_meta, SQL_INS_CHART_LABEL, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement store chart labels");
+        return;
+    }
+
+    rc = sqlite3_bind_blob(res, 1, chart_uuid, sizeof(*chart_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind chart_id parameter to store label information");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int(res, 2, source_type);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind type parameter to store label information");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_text(res, 3, label, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind label parameter to store label information");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_text(res, 4, value, -1, SQLITE_STATIC);
```

Comment:
can we have empty values?

Reply:
looks like it can be an empty string indeed but should not be NULL....

Reply:
Well (for example) the `parse_k8s_data` function will not allow empty values. 

However there is no problem in the `value` being null here. Actually a null value binds to null properly and stores a null in the database in that case. 

So value = NULL would be equivalent to `sqlite3_bind_null(res, 4)`

Reply:
what I have in mind is "" -> NULL in DB

Reply:
The "" will be valid and stores a value of size 0 in the database. 

For the `create_label` maybe we need to recheck but most likely the caller does the extra checks needed (so it does not fail in unexpected ways)